### PR TITLE
ADEN-8352  Introduce Typescript

### PR DESCRIPTION
**Pull request by Bielik20**
Important note: Babel was updated to 7.

Branch made from ADEN-8532 following the below instructions for painless merge.
```
# update dev branch
git checkout dev
git pull

# rename files
find spec -depth -execdir rename 's/spec.js/spec.ts/' '{}' \;
find src -depth -execdir rename 's/\.js/\.ts/' '{}' \;     
find examples -depth -execdir rename 's/script.js/script.ts/' '{}' \;
mv src/ad-services/nielsen/static-queue-script.ts src/ad-services/nielsen/static-queue-script.js
find . -depth -execdir rename 's/.tson/.json/' '{}' \;
find src -depth -execdir rename 's/.tson/.json/' '{}' \;

# stage changes
git status
git add .

# create a WIP commit
git commit -m "rename .js -> .ts"

# rebase ADEN-8532 branch accepting theirs when you did not make any changes to file
git checkout ADEN-8352
git pull
git rebase dev --interactive # (squash all commits to one)

# squash the WIP commit with the main commit, amend commit messages if necessary
git pull origin dev --rebase=interactive #squash "rename .js -> .ts" commit with the main commit

# IF everything was smooth, push your work
git push --force-with-lease

# reset your dev branch
git checkout dev
git reset --hard HEAD~1
```

### CODE REVIEW

**Comment by Bielik20** [comment]
Meaningful files:
-  examples/video/jwplayer/script
- .babelrc
- .eslintignore
- package.json
- tsconfig.json
- wdio.conf.js
- webpack.config.js

File: .babelrc [hunk 1]
```diff
@@ -1,35 +1,58 @@
...
-		"transform-object-rest-spread",
-		"transform-decorators-legacy",
+		[
+			"@babel/plugin-proposal-decorators",
+			{
+				"legacy": true
+			}
+		],
+		[
+          "@babel/plugin-proposal-class-properties",
```
Comment (bckatarzyna) [inline .babelrc]: There are spaces instead of tabs here

File: package.json [hunk 2]
```diff
@@ -102,14 +104,16 @@
 		"version": "./maintenance/version.sh",
 		"serve": "./node_modules/webpack-dev-server/bin/webpack-dev-server.js --progress --colors --inline --open --watch --content-base=./examples",
 		"tdd": "mocha-webpack -w --full-trace './spec/**/*.spec.js' -r jsdom-global/register --webpack-env.test",
```
Comment (bckatarzyna) [inline package.json]: I'm not sure but I want to ask: Don't we also need .ts here?

File: .babelrc (hunk 1 above)
Comment (Bielik20) [reply .babelrc]: No, they are not. If they were the "plugins" would have been moved as well.

File: package.json (hunk 2 above)
Comment (Bielik20) [reply package.json]: Fixed

File: .eslintignore [hunk 3]
```diff
@@ -4,6 +4,7 @@ lib/
 node_modules/
 jspm_packages/
 tests/allure-results
+src/
```
Comment (fraszczakszymon) [inline .eslintignore]: ? :D

Comment (Bielik20) [reply .eslintignore]: Wyleci jak dodam TSLint.

**Pull request merged**